tests/models/decoder_only/vision_language/test_models.py

```diff
@@ -0,0 +1,629 @@
+from .vlm_utils.case_filtering import get_parametrized_options
+from .vlm_utils.core_utils import (run_custom_inputs_test, run_embedding_test,
+                                   run_multi_image_test, run_single_image_test,
+                                   run_video_test)
+from .vlm_utils.mm_test_types import (CustomTestOptions, ImageSizeWrapper,
+                                      VLMTestInfo, VLMTestType)
+
+# This hack is needed for phi3v & paligemma models
+# ROCm Triton FA can run into shared memory issues with these models,
+# use other backends in the meantime
+# FIXME (mattwong, gshtrasb, hongxiayan)
+if is_hip():
+    os.environ["VLLM_USE_TRITON_FLASH_ATTN"] = "0"
+
+COMMON_BROADCAST_SETTINGS = {
+    "test_type": VLMTestType.IMAGE,
+    "dtype": "half",
+    "max_tokens": 5,
+    "tensor_parallel_size": 2,
+    "image_size_factors": ((.25, 0.5, 1.0), ),
+    "distributed_executor_backend": (
+        "ray",
+        "mp",
+    ),
+    "skip": cuda_device_count_stateless() < 2,
+}
+
+### Test configuration for specific models
+# NOTE: The convention of the test settings below is to lead each test key
+# with the name of the model arch used in the test, using underscores in place
+# of hyphens; this makes it more convenient to filter tests for a specific kind
+# of model. For example....
+#
+# To run all test types for a specific key:
+#     use the k flag to substring match with a leading square bracket; if the
+#     model arch happens to be a substring of another one, you can add a
+#     trailing hyphen. E.g.,
+#                 - pytest $TEST_FILE -k "[llava-"
+#     prevents matching on "[llava_next-" & will match just the enabled cases
+#     for llava, i.e., single image, image embedding, and custom input tests.
+#
+# To run a test for a Test Info for just one of multiple models:
+#     use the k flag to substring match the model name, e.g.,
+#                 - pytest $TEST_FILE -k OpenGVLab/InternVL2-1B
+#     prevents matching on nGVLab/InternVL2-2B.
+#
+# You can also combine substrings to match more granularly.
+#     ex 1:
+#        pytest $TEST_FILE -k "test_single_image and OpenGVLab/InternVL2-1B"
+#     will run only test_single_image* for OpenGVLab/InternVL2-1B; this would
+#     match both wrappers for single image tests, since it also matches
+#     test_single_image_heavy (which forks if we have a distributed backend)
+#     ex 2:
+#        pytest $TEST_FILE -k  "[llava- or [intern_vl-"
+#     will run all of the tests for only llava & internvl.
+#
+# NOTE you can add --collect-only to any of the above commands to see
+# which cases would be selected and deselected by pytest. In general,
+# this is a good idea for checking your command first, since tests are slow.
+#
+# Lastly, be aware keep in mind that this will not show tests if their skip
+# condition is met since they're filtered and won't be picked up by pytest,
+# e.g., trying to run the broadcast tests in a non-distributed environment.
```

Comment:
Imo showing the skipped tests via pytest is more informative than skipping the creation of the test decorator, so at least you know that the test has been successfully registered.

Reply:
Agreed. I tweaked the filter a bit to wrap all the expanded cases in pytest parameters so that tests that match, but whose skip conditions are met, correctly get registered but skipped by pytest instead of getting dropped before pytest gets to them